packages/pwa/app/components/_app-config/index.jsx

```diff
@@ -83,10 +97,15 @@ const AppConfig = ({children, locals = {}}) => {
 
 AppConfig.restore = (locals = {}) => {
     // Parse the locale from the page url.
+    const {originalUrl} = locals
     const locale = getLocale(locals) || DEFAULT_LOCALE
+    const basePath = originalUrl !== HOME_HREF ? getLocaleAndSiteBasePath(locale) : undefined
+    console.log('basePath', basePath)
+
     const currency = getPreferredCurrency(locale) || DEFAULT_CURRENCY
 
     locals.api = new CommerceAPI({...apiConfig, locale, currency})
+    locals.basePath = basePath
```

Comment:
Hey @bendvc I am totally aware the fact that we want to shy away from using `locals`. I am wondering is there any other good place to pass `basePath` to the server-side? 🤔 

My thought on this: since we are still debating on how we want to replace the `locals` with another api, I did not want to spend my time on that as the main point here is to explore the idea of whether we can make use of the `basename` on the Router for the url customisation.

Reply:
Had a short chat with @bendvc, just want to point out that using `locals` to pass basePath to SSR is not the correct way to do it. We are still figuring out a better way to do so (to name a few PR for that #188 , #184 ). If the solution is chosen, we will need to fix the logic to adapt the change obviously. 
In this PR, I just want to explore the idea of accepting the basename prop on the SDK Routers and generate the routes dynamically without modifying the routes.jsx in the pwa.  To explain how it work from an example:

The  [basename](https://reactrouter.com/web/api/StaticRouter/basename-string) acts as a base path for all the routes. Imagine we have a routes array like this:
```
const routes = [
   {
        path: '/',
        component: Home,
        exact: true
    },
    {
        path: '/product/:productId',
        component: ProductDetail
    },
]
```
Let's say we have a config file say we want the `locale` to be included in the path

```
"url": {
          "locale": "path"
        },
```
This means we want to `locale` to be the baseUrl for all routes. When we access the `/en-GB/product/123`, it will match the ` '/product/:productId'` without having us clarify the param in the path explicitly (inside routes.jsx in PWA) because it has been handled from the SDK side. 

The logic to generate basePath is determined by `getLocaleAndSiteBasePath` which will look into the `url` config and return the `basePath` output.